src/transmission_manager.rs

```diff
@@ -44,36 +50,38 @@ fn evaluate_transmission(context: &mut Context, contact_id: PersonId, _transmitt
 
 fn infection_attempt(
     context: &mut Context,
-    transmitee_id: PersonId,
-    gi: f64,
-    num_infection_attempts_remaining: f64,
+    transmitter_id: PersonId,
+    num_infection_attempts_remaining: usize,
     next_infection_time_unif: f64,
 ) -> Result<(), IxaError> {
-    // if the person still has people left to infect
-    if num_infection_attempts_remaining >= 1.0 {
-        // this is a method from a trait extention implemented in the contact manager
-        // as long as this method returns a contact id, it can use any underlying sampling strategy
-        let contact_id = context.get_contact(transmitee_id)?;
-        // evaluate transmission is its own function because there will be
-        // intervention-based logic there eventually
-        // if there are no contacts, to infect, do nothing
+    // Only attempt the infection if the person has infection attempts remaining.
+    if num_infection_attempts_remaining >= 1 {
```

Comment:
```suggestion
    if num_infection_attempts_remaining > 0 {
```
This is now an integer.